From 3fd5cd5a82b7972d4a01d5cd4ec52c5f82fe9a33 Mon Sep 17 00:00:00 2001
From: Stuart Menefy <stuart.menefy@codasip.com>
Date: Tue, 16 Jan 2024 10:42:23 +0000
Subject: [PATCH 05/12] spi: xilinx_spi: Fix startup workaround

The call to xilinx_spi_startup_block() in xilinx_spi_xfer() needs to
occur before the CS is asserted for the real transfer, otherwise the
workaround will break the first real transfer.

Signed-off-by: Stuart Menefy <stuart.menefy@codasip.com>
Fixes: 6c9268650 ("spi: xilinx_spi: Reinstate xfer function")
---
 drivers/spi/xilinx_spi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/xilinx_spi.c b/drivers/spi/xilinx_spi.c
index 7eb7b152a0..f826d03ee2 100644
--- a/drivers/spi/xilinx_spi.c
+++ b/drivers/spi/xilinx_spi.c
@@ -313,11 +313,11 @@ static int xilinx_spi_xfer(struct udevice *dev, unsigned int bitlen,
 		goto done;
 	}
 
+	xilinx_spi_startup_block(dev);
+
 	if (flags & SPI_XFER_BEGIN)
 		spi_cs_activate(dev, slave_plat->cs);
 
-	xilinx_spi_startup_block(dev);
-
 	start_transfer(dev, dout, din, bytes);
 
  done:
-- 
2.39.3

